library/ssl_tls.c

```diff
@@ -8259,20 +8259,27 @@ int mbedtls_psa_ecjpake_write_round(
             step <= PSA_PAKE_STEP_ZK_PROOF;
             ++step )
         {
-            /* For each step, prepend 1 byte with the length of the data */
-            *(buf + output_offset) = MBEDTLS_SSL_ECJPAKE_OUTPUT_SIZE( step );
-            output_offset += 1;
-
+            /*
+             * For each step, prepend 1 byte with the length of the data.
+             *
+             * NOTE = psa_pake_output() sometimes output elements which are
+             * NOT 32 or 65 bytes as expected, but 1 byte less. So, instead
```

Comment:
I would rather suggest to explain the reasons, not only results of this :) Why does this happen? Is there a leading zero trimmed (by what function)? Can it be more zeros?